examples/ex33.hpp

```diff
@@ -0,0 +1,318 @@
+   DenseMatrixGeneralizedEigensystem eig1(E,B);
+   eig1.Eval();
+   Vector & evalues = eig1.EigenvaluesRealPart();
+   for (int i = 0; i<evalues.Size(); i++)
+   {
+      if (IsFinite(evalues(i)))
+      {
+         poles.Append(evalues(i));
+      }
+   }
+#else
+   mfem_error("Compiled without LAPACK");
+#endif
+   // compute the zeros
+   B = 0.;
+   E = 0.;
+   for (int i = 1; i<=m; i++)
+   {
+      B(i,i) = 1.;
+      E(0,i) = w(i-1) * f(i-1);
+      E(i,0) = 1.;
+      E(i,i) = z(i-1);
+   }
+
+#ifdef MFEM_USE_LAPACK
+   DenseMatrixGeneralizedEigensystem eig2(E,B);
+   eig2.Eval();
+   evalues = eig2.EigenvaluesRealPart();
+   for (int i = 0; i<evalues.Size(); i++)
+   {
+      if (IsFinite(evalues(i)))
+      {
+         zeros.Append(evalues(i));
+      }
+   }
+#else
+   mfem_error("Compiled without LAPACK");
+#endif
+
+   scale = w * f / w.Sum();
+}
+
+
+void PartialFractionExpansion(double scale, Array<double> & poles,
+                              Array<double> & zeros, Array<double> & coeffs)
+{
+   int psize = poles.Size();
+   int zsize = zeros.Size();
+   coeffs.SetSize(psize);
+   coeffs = scale;
+
+   for (int i=0; i<psize; i++)
+   {
+      double tmp_numer=1.0;
+      for (int j=0; j<zsize; j++)
+      {
+         tmp_numer *= poles[i]-zeros[j];
+      }
+
+      double tmp_denom=1.0;
+      for (int k=0; k<psize; k++)
+      {
+         if (k != i) { tmp_denom *= poles[i]-poles[k]; }
+      }
+      coeffs[i] *= tmp_numer / tmp_denom;
+   }
+}
+
+
+// x^-a
+void ComputePartialFractionApproximation(double alpha,
+                                         Array<double> & coeffs, Array<double> & poles,
+                                         double lmin = 1., double lmax = 1000.,
+                                         double tol=1e-10,  int npoints = 1000,
+                                         int max_order = 100)
+{
+#ifndef MFEM_USE_LAPACK
+   mfem::out
+         << "MFEM is compiled without LAPACK. Using precomputed PartialFractionApproximation"
+         << std::endl;
+
+   if (alpha == 0.33)
+   {
+      coeffs = Array<double> ({2002.55,  99.7691,  29.0575,  12.9842,
+                               6.90263,  3.96688,  2.36013,  1.42565,
+                               0.867623, 0.529436, 0.317975, 0.0891797});
+      poles = Array<double> ({-47928.,  -3451.45, -996.636, -388.58,
+                              -168.228, -75.3162, -33.6235, -14.5403,
+                              -5.84563, -1.9975, -0.434608,  0.});
+   }
+   else if (alpha == 0.99)
+   {
+      coeffs = Array<double>({0.0292386, 0.0143338, 0.0109015, 0.00976958,
+                              0.00943417, 0.00948077, 0.00985799, 0.010889,
+                              0.0138588, 0.0263289, 0.96953});
+      poles = Array<double> ({-10085.1, -1652.6, -524.342, -199.521,
+                              -80.2328, -32.3926, -12.7148, -4.63855,
+                              -1.39884, -0.221076, 0.});
+   }
+   else
+   {
+      if (alpha != 0.5)
```

Comment:
Why these values are used by default? Aren't they valid for 0.5? If so just write in the documentation that the code is implemented for 0.33, 0.99 and 0.5.

Reply:
Done